### ISSUE TITLE
More details on how it works

### ISSUE THREAD

**Issue by charleschege** [PaulGrandperrin/XChaCha20Blake3Siv#1]
I am not a cryptography expert but I have asked the same question on whether we can change Chacha20-poly1305 to use Blake3. It seems that you have come up with a way to do this. I read your comments on blake3 repo https://github.com/BLAKE3-team/BLAKE3/issues/138 so why not publish the details here in this repo

**Comment by cryptoquick** [comment]
Yeah, I'd welcome some more details also. This sounds heckin cool! :rocket:
